lib/ts/recipe/session/sessionRequestFunctions.ts

```diff
@@ -325,6 +326,28 @@ export async function refreshSessionInRequest({
         }
         throw ex;
     }
+
+    // TODO-SAN: should we check for anomaly detection here, what about performance?
+    const anomalyDetection = AnomalyDetectionRecipe.getInstance();
+    if (anomalyDetection !== undefined) {
+        const anomalyDetectionResult = await anomalyDetection.recipeInterfaceImpl.checkAnomalyWithRequest({
+            request: req,
+            // TODO-SAN: assuming status is OK, should we check?
+            status: "OK",
+            userId: session.getUserId(),
+            tenantId: session.getTenantId(),
+            action: "REFRESH_TOKEN",
+        });
+        if (anomalyDetectionResult.status === "ANOMALY_DETECTED") {
+            // TODO-SAN: should we throw exception or return status not OK?
```

Comment:
throwing an exception is the right thing to do here.

Reply:
ok